### PR TITLE
Make sure we use UTF-8 support from R 4.2.1 on windows

### LINKED ISSUE #1758: [Bug]: special character in user name results in engine crash with plots
### JASP Version

0.16.3

### Commit ID

_No response_

### JASP Module

jaspDescriptives, jaspRegression, jaspTTests

### What analysis are you seeing the problem on?

Any plot in any module

### What OS are you seeing the problem on?

Windows 11

### Bug Description

Hi everybody,

all of my calculations in JASP work fine as long as I don't use any plots. As soon as I chosse to display descriptive plots or raincloud plot in T-Tests,  distribution plot in descriptives and so on, it will display a white window with heading and as soon as I try to scale it, the error message is dropped "The engine crashed while trying to run this analysis". 
Even more interesting: When I add a second type of plot, the first one will eventually display, sometimes for all of the items, sometimes for 1/4, however, again as soon as I try doing anything with it, the error message drops again. Tried my own data file and also the famous "kitchen rolls" testfile, same error.
I've read similar problems here already, but the solutions did not work for me (Region settings for instance).

I'd be glad about any help!

PS: First occurred to me in 0.16.2, now I was hoping to solve it with 0.16.3, which is not the case.

### Expected Behaviour

In versions prior to 0.16.2 plots would always display correctly and don't drop any error message when e.g. trying to scale them

### Steps to Reproduce

1. Load any variable in any test,
2. Check any analysis that is a plot,
3. Scale the window or try "edit image" and hit "ok"
...

### Log (if any)

[JASP 2022-06-22 12_27_15 Desktop.log](https://github.com/jasp-stats/jasp-issues/files/8956759/JASP.2022-06-22.12_27_15.Desktop.log)
[JASP 2022-06-22 12_27_15 Engine 0.log](https://github.com/jasp-stats/jasp-issues/files/8956760/JASP.2022-06-22.12_27_15.Engine.0.log)
Screenshot 2022-06-22 122627

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if any.
- [X] I have included a JASP file that causes the crash/bug, if any.
- [X] I have accurately described the bug, and steps to reproduce it.

### PR DESCRIPTION
**Pull request by JorisGoosen**
Fixes https://github.com/jasp-stats/jasp-issues/issues/1758 (engine crash on nonlatin characters in username on windows)

ps. All the nativeToUtf8 workaround functions can also be removed, but there is no hurry

### CODE REVIEW

**Comment by JorisGoosen** [comment]
And also, some of the changes mightve become obsolete due to the active codepage.
But on the other hand, Id like to keep JASP.exe running in local codepage so that we can still support loading chinese-locale csvs in JASP and such shenanigans

File: Engine/main.cpp [hunk 1]
```diff
@@ -109,14 +109,18 @@ int main(int argc, char *argv[])
 
 		Engine e(0, 0); //It needs to start to make sure rbridge functions work
 
+#ifdef _WIN32
+	_m	oduleLibraryFixer(Utils::wstringToString(argv[1]), true, true);
```
Comment (boutinb) [inline Engine/main.cpp]: oeps

File: R-Interface/jasprcpp.cpp [hunk 2]
```diff
@@ -98,7 +101,10 @@ void STDCALL jaspRCPP_init(const char* buildYear, const char* version, RBridgeCa
 	_stringNativeToUtf8		= nativeToUtf8;
 
 	jaspRCPP_logString("Creating RInside.\n");
+
 	rinside = new RInside();
+
+	R_TempDir = (char*)tempDir;
```
Comment (boutinb) [inline R-Interface/jasprcpp.cpp:107]: Don't understand: this variable does not seem to be used.

**Review by boutinb — APPROVED** [review]

File: R-Interface/jasprcpp.cpp (hunk 2 above)
Comment (JorisGoosen) [reply R-Interface/jasprcpp.cpp:107]: If you check where it is defined you'll see it is `extern` and it is in fact in R.
There were some issues on windows where the environment variables were converted to local codepage... Which messed up stuff if it couldnt fit. So I now just worked around it like this.

But actually, I think setting the active codepage to utf-8 probably already solved that. 

However, it works now and we can always clean stuff up later.

Comment (JorisGoosen) [reply R-Interface/jasprcpp.cpp:107]: Previously this value was shared through `TMPDIR`

**Pull request merged**